pkg/cmd/roachtest/test_runner.go

```diff
@@ -931,16 +931,9 @@ func (r *testRunner) runWorker(
 				}
 
 				// 50% chance of enabling the rangefeed buffered sender. Disabled by
-				// default. This is the reason we check for run time assertions rather
-				// than using rand:
-				// 1. The cluster setting RangefeedUseBufferedSender validation will fail
-				// if buildutil.CrdbTestBuild tag is not set.
-				// 2. There is a 50% chance that the cockroach binary used in roachtests
-				// includes the buildutil.CrdbTestBuild tag. And runtime assertion is
-				// enabled if and only if cockroach binaries used in roachtests include
-				// the buildutil.CrdbTestBuild tag.
-				useBufferedSender := roachtestutil.UsingRuntimeAssertions(t)
-				if useBufferedSender {
+				// default.
+				useBufferedSender := prng.Intn(2) == 0
+				if t.IsBuildVersion("v25.2.0") && useBufferedSender {
```

Comment:
Should we do `t.BuildVersion().AtLeast()` instead so this check doesn't break when we bump the version? Although I'm not sure if either check fixes mixed version tests. Unfortunately the build version referred to here doesn't have any knowledge of the  binaries the mixed version frame work uploads, it's essentially just a check on the version the branch is on.

Maybe the correct fix should be for the mixed version framework to ignore `c.clusterSettings`? Seems like that will always break whenever a new setting is added.

Reply:
Thanks for taking a look! 

> Should we do t.BuildVersion().AtLeast() instead so this check doesn't break when we bump the version?

https://github.com/cockroachdb/cockroach/blob/3ca9845cc23d325eafd8977d5ce76c68b6ace89d/pkg/cmd/roachtest/test_impl.go#L692 

`t.IsBuildVersion` checks for `AtLeast()`. 

> Unfortunately the build version referred to here doesn't have any knowledge of the binaries the mixed version frame work uploads, it's essentially just a check on the version the branch is on.

You are right. Just noticed this from the failing CI 😢. Will try to look into this further and figure out the next steps.